src/metabase/api/collection.clj

```diff
@@ -91,30 +91,50 @@
        (sort-by (comp str/lower-case :name))))
 
 (s/defn ^:private collection-detail
-  "Add a standard set of details to `collection`, including things like `effective_location` as well as items in the
-  Collection, restricted by `children-options`. Works for either a normal Collection or the Root Collection."
-  [collection :- collection/CollectionWithLocationAndIDOrRoot, children-options :- CollectionChildrenOptions]
+  "Add a standard set of details to `collection`, including things like `effective_location`.
+  Works for either a normal Collection or the Root Collection."
+  [collection :- collection/CollectionWithLocationAndIDOrRoot]
   (-> collection
-      (hydrate :effective_location :effective_ancestors :can_write)
-      (assoc :items (collection-children collection children-options))))
+      (hydrate :effective_location :effective_ancestors :can_write)))
+
+(s/defn ^:private collection-items
+  "Return items in the Collection, restricted by `children-options`.
+  Works for either a normal Collection or the Root Collection."
+  [collection :- collection/CollectionWithLocationAndIDOrRoot, children-options :- CollectionChildrenOptions]
+  (collection-children collection children-options))
 
 (api/defendpoint GET "/:id"
-  "Fetch a specific Collection with the following options:
+  "Fetch a specific Collection with standard details added"
+  [id]
+  (collection-detail (api/read-check Collection id)))
+
+(api/defendpoint GET "/:id/items"
+  "Fetch a specific Collection's items with the following options:
 
   *  `model` - only include objects of a specific `model`. If unspecified, returns objects of all models
   *  `archived` - when `true`, return archived objects *instead* of unarchived ones. Defaults to `false`."
   [id model archived]
   {model    (s/maybe (s/enum "card" "dashboard" "pulse" "collection"))
    archived (s/maybe su/BooleanString)}
-  (collection-detail
+  (collection-items
     (api/read-check Collection id)
     {:model     (keyword model)
      :archived? (Boolean/parseBoolean archived)}))
 
-
 ;;; -------------------------------------------- GET /api/collection/root --------------------------------------------
 
 (api/defendpoint GET "/root"
+  "Return the 'Root' Collection object with standard details added"
+  []
+  (-> (collection-detail collection/root-collection)
+      ;; add in some things for the FE to display since the 'Root' Collection isn't real and wouldn't normally have
+      ;; these things
+      (assoc
+          :name (tru "Saved items")
+          :id   "root")
+      (dissoc :metabase.models.collection/is-root?)))
```

Comment:
I like `::collection/is-root?` here instead. That way if we decide to rename the namespace in the future, `is-root?` will automatically get updated (otherwise we get a compilation error). As-is, if we move the ns and forgot to rename it here, it will just try to remove a key that doesn't exist.